### PR TITLE
chore (docs): improve example content, highlighting & install snippets

### PR DESCRIPTION
**Pull request by agusmoles**
Hi team, I was going through the documentation of the Vercel AI SDK for RSC and I think these changes were wrong in the original documentation, as we were obtaining the data from the `getWeather` external function, destructuring the `content` but then not using the returned variables since the attributes are being hardcoded in the `return` statement of each one

Let me know if I got something wrong. Great tool and documentation btw! 😄 

EDIT: made some more updates, described below (and in the commit msgs)

### CODE REVIEW

File: content/docs/04-ai-sdk-rsc/01-overview.mdx [hunk 1]
```diff
@@ -175,17 +175,9 @@ const text = generateText({
...
         uiStream.done(
-          <WeatherCard
-            weather={{
-              temperature: 47,
-              unit: 'F',
-              description: 'sunny'
-              forecast,
-            }}
-          />
+          <WeatherCard weather={weather} />
```
Comment (lgrammel) [inline content/docs/04-ai-sdk-rsc/01-overview.mdx:180]: this will affect the highlighting (line 158) which needs to be updated

**Comment by lgrammel** [comment]
thank you!

File: content/docs/04-ai-sdk-rsc/01-overview.mdx (hunk 1 above)
Comment (agusmoles) [reply content/docs/04-ai-sdk-rsc/01-overview.mdx:180]: oh you are right, fixed

**Comment by agusmoles** [comment]
I've also added a fix to remove an unused improt that might creates confusion (at least it did for me) about if that was needed or not and also typed the `weather` variable

**Comment by agusmoles** [comment]
Added tabs to install the dependencies with the different package managers (to be consistent with other parts of the documentation like [https://sdk.vercel.ai/docs/ai-sdk-core/schemas-and-zod](https://sdk.vercel.ai/docs/ai-sdk-core/schemas-and-zod))

**Review by lgrammel — APPROVED** [review]
Thank you, this is very helpful!

**Pull request merged**